Adds medibeam to all Syndicate modes.[READY][TESTED]

lordsaladin opened the pull request:
[Changelogs].

:cl: lord saladin(ckey)
add: Adds medibeam gun to traitor uplinks.
Tweak: Now costs 10 TC again(if you die to two traitors using medibeam guns then it's your fault.) and has a chance to spawn in surplus and null crates, Making it more inline with Traitor modes, restricted to MD/CMO otherwise.
Update: now has correct path.
/:cl:

[why]: Want to feel like your a Odysseus or be a friendly traitor, now you can as this pr adds the medibeam gun to uplinks. NOTE yorii has tested and it works.

Commit:
Adds medibeam to traitor modes.

GuppyLaxx commented:
Let me stop you right there and say........ **Friendly traitors are gay.**
Apart from that, 👍 I'm expecting some funny gimmicks

Jarektheraptor commented:
Time to cross the beams

Commit:
Reduced cost plus added surplus chance.

In order to allow some breathing room and to make it more compatible with traitor modes, I've added a surplus chance and reduced the cost now you can also grab a emag and a syndie key or other gear.

lordsaladin commented:
Because every game needs a ranged medic.

Cyprex commented:
I suggest just removing the gamemode requirement instead.

GuppyLaxx commented:
10 TC sounds like 2 greytides rushing in and medibeaming the captain at the same time

CitrusGender commented:
isn't this what you just did in your last PR that got closed except you're adding it to traitor mode

Commit:
Removed game mode requirements.

lordsaladin commented:
@CitrusGender  This grew out of that idea, less memey now.
Was told to make a separate pr for this.

CitrusGender commented:
ok now it's just in all modes

aight

Cyprex commented:
Their previous PR changed various surplus chances for null crates.

lordsaladin commented:
Well it's intent was to make sure you get trolled by rng less,

CitrusGender commented:
yeah the surplus chance is changed in this one too for just medibeams now (in addition to the other Prs)

lordsaladin commented:
@CitrusGender  the last one was closed.

GuppyLaxx commented:
last one changed surplus values of a ton of shit WHILE adding medbeams to traitors

lordsaladin commented:
@Spaceinaba  They overproduced them and they want to encouraged team work,Also blood brothers exist.

Spaceinaba commented:
Blood brothers don't have TC last I checked.

lordsaladin commented:
@Spaceinaba Ok they wanted to encourage betrayal,Also y'all keep saying be the change the world needs so this is that change.(plus sec will give it to medical if you are caught with it as it is clearly not a weapon.)

PKPenguin321 commented:
friendly traitors suuuuck

lordsaladin commented:
@PKPenguin321 Have you ever played a role where having a friendly traitor is handy, or are you a greytide main?

Wilchenx commented:
Medbeams cost 10 tc now


Can't wait to see traitor duos cross beams on people in hilarious murder spree.

lordsaladin commented:
That takes more teamwork then le desword no slip muderbones.

davethwave commented:
Might make more sense to make it a medical only item.

lordsaladin commented:
@davethwave  nah then it would be another job specific item and i don't want that.

PKPenguin321 commented:
@lordsaladin fuck you, "if you disagree with me u r le greytide main." antagonists are there to provide conflict in the round and make it actually fucking meaningful rather than just the crew grinding out all the same research/botany/etc that it did the same way the round before. traitors that are explicitly "friendly" and go out of their way to not only not antagonize but stop other antagonists actively disrupt the conflict of the round and make the game boring as fuck. if your only comeback is that i'm a """greytide main""" (i'm not, and what the fuck does that even have to do with anything, you absolute retard), you can fuck yourself.

Bawhoppen commented:
Absolutely no reason for this change. An OP boring item like this that only makes sense in the context of a team mode like nukeops definitely shouldn't be added to standard traitor. 10TC is also way way too cheap.
At best, you might find it reasonable to add as a job specific traitor item for doctors, otherwise this is a big fat :-1: from me.

lordsaladin commented:
@PKPenguin321 wow iirc Github have explicitly told us not to swear on here in the past,Have you not played terry or lowpop rounds if the antag goes SSD or dies early it basically becomes extended and we can get stuff done, Don't forget betrayal exists you can just cause mayhem when they least expect it.

Commit:
Now costs 20 TC because it's clearly boring and op you should realy build an Odysseus ,also resticted to CMO and MD because y'all can't be nice, why am i even doing this for you ungrateful jerks.

lordsaladin commented:
THERE YOU UNGRATEFUL BEEPS NOW IT COSTS ALL YOUR TC.

GLACoding commented:
Can you remove the surplus chance?
Its a pretty costly and extremely niche item that would basically be useless for a normal traitor which would dilute the overall value of his surplus drop if he was unlucky enough to get it.

lordsaladin commented:
@GLACoding ok then i'll remove all the other  restricted items from surplus crates no more brain washing  or loyal borgs or you!

CitrusGender commented:
here we go

lordsaladin commented:
All this burning my eyes on github's website trying to implement a idea for y'all to salt on it,you wouldn't like it if i did that to your prs.

CitrusGender commented:
tweaks/balances get scrutiny since they're all just opinions. This isn't new.

Commit:
Just let me have my two cents, I could be playing /tg/ right now.

optimumtact commented:
if you are this sensitive about people giving you critical feedback you will not last very long here at all.

Try to seperate peoples comments on your change with their comments on you as a person or you will quickly burn out

ExcessiveUseOfCobblestone commented:
That and stop making a diff pull request lol

kevinz000 commented:
is this tested

Tlaltecuhtli commented:
make the medibeam have a second mode where it does negative healing

Tlaltecuhtli commented on code/modules/uplink/uplink_items.dm:
```diff
@@ -1149,9 +1149,10 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 	desc = "A wonder of Syndicate engineering, the Medbeam gun, or Medi-Gun enables a medic to keep his fellow \
 			operatives in the fight, even while under fire."
```
operatives? more like metafriends

Tlaltecuhtli reviewed (changes requested):
also look at the end of file the restricted roles items have a different path

Iatots commented:
Are you from terry?

lordsaladin commented:
Terry. yes
update Yorii has tested and it works.

Noahnoah55 commented:
I wonder if you can use a gun firing circuit with the medigun. Sounds like it could get very broken very quickly.

Commit:
Update uplink_items.dm

Tlaltecuhtli commented on code/modules/uplink/uplink_items.dm:
```diff
@@ -1147,11 +1147,12 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 /datum/uplink_item/device_tools/medgun
```
needs to be 
/datum/uplink_item/role_restricted/medgun

epochayur commented:
you can't go making random balance changes for no apparent reason and not expect to get some backlash

improvedname commented:
I guess i can get syndicate borgs and medibeams trough null crates noe as a non antag

iskyp commented:
what @epochayur said except you're not doing this for kicks

Commit:
fuxed

Commit:
path corrected.

AlexTheSysop42 commented:
@improvedname first, null crates give op gear during op rounds so maybe

Second, non-ops can't use syndie spawners

ExcessiveUseOfCobblestone commented:
@AlexTheSysop42 he's changing the medbeam to be not op specific, so you'd be able to get medbeams in nulls and surplus post this change (unless he sets `surplus_nullcrates` to 0).

AlexTheSysop42 commented:
He specifically mentioned syndicate borgs

improvedname commented:
syndicate borgs is how you get a emagg as a non antag

lordsaladin commented:
Oh wow several admins dislike this?
Geez let me have my 2 minutes of fame on the changelog.

iskyp commented:
@lordsaladin this is not how you get people to agree with your point

epochayur commented:
![untitled](https://user-images.githubusercontent.com/32695675/40945957-df5ffa5a-680f-11e8-8e4f-5cbef6bee31e.png)
jesus christ

iskyp commented:
\> +1's their own PR
hahahahahahahahahahahahahahahahahahahahahahahahahahaha

davethwave commented:
It would be interesting if you can at least get medibeams in null crates

lordsaladin commented:
@davethewave this adds that.

ExcessiveUseOfCobblestone commented:
@lordsaladin just fyi admins have 0 power here, and headmins only have say out of communal interest between admins/code (features would be pretty useless if you just got banned for using them 🤔).

However it seems like a nice bit of players dislike the PR as well.

lordsaladin commented:
Of course they do this involves surplus/null crates.

ShizCalev commented:
![image](https://user-images.githubusercontent.com/6209658/41134493-9bfe6142-6a9a-11e8-86c8-af8fea8833bb.png)

lordsaladin commented:
@ShizCalev That not a valid reason or are you apart of the surplus/null crate hate cabal, like the rest of the people that disliked that want to deny me of having my name on the changelog.

PKPenguin321 commented:
You need to not take things personally and to stop acting so entitled

iskyp commented:
@lordsaladin its very easy to get your name on the channeling, just make a fix. you're not entitled to anything just because you think your changes are better for the game

also, I'm betting an entire bag of popcorn that half of the people who -1d this probably did it because you had a temper tantrum whenever you were told this was a bad change

ExcessiveUseOfCobblestone commented:
📸 ADD 📸 ME 📸 TO 📸 THE 📸  SCREENSHOT 📸

Armhulen commented:
:camera_flash: Yep, this one's going into my usual github activity compilation.

CitrusGender commented:
based shiz
